### PR TITLE
Patch consolidation for NCL recipe

### PR DESCRIPTION
**Pull request by vanderwb**
closes #43280

This PR consolidates the two `patch()` functions, since this duplication was causing the csh patch to be lost. I've also moved the other csh shebang filters to the patch so that they will be applied if `spack patch` is run.

This PR supersedes #43280, which attempted to fix the multiple-patch issue via a `run_before` decorator on the first patch.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Waiting for @AlexanderRichert-NOAA before merging

**Comment by alexrichert** [comment]
This looks good to me, please merge :)

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Comment by vanderwb** [comment]
@alalazo - I think we have all needed approvals. Thanks.

**Pull request merged**
